src/BusinessLayer/TelemetryReporting.h

```diff
@@ -33,29 +35,58 @@ class TelemetryReporting : public QObject
                        const LightsData& lightsData_,
                        View& view);
 
+    friend class TelemetryReportingTest;
```

Comment:
We shouldn't need to call private functions, even in our tests. By adding code in here specifically for testing, we are also hurting the portability of this code. If we ever wanted to move it somewhere, google test would have to be added to that project. 

Testing should be done by using the entirety of the public interface, doing this should test all the private functionality as well. When the mocked Data classes are passed into the constructor of this class, this will test the private slots because TelemetryReporting does the job of connecting them.

For the private functions like framing data and stuff, this was a design flaw from before you touched this project (i.e. my fault ;D). All the private functions are too much responsibility for this one class and should be moved to a different class where the functions can be public and properly tested.